internal/allocation/allocation.go

```diff
@@ -80,70 +85,50 @@ func (a *Allocation) RemovePermission(fingerprint string) {
 func (a *Allocation) AddChannelBind(c *ChannelBind, lifetime time.Duration) error {
 	// Check that this channel id isn't bound to another transport address, and
 	// that this transport address isn't bound to another channel id.
-	channelByID := a.GetChannelByID(c.ID)
-	channelByPeer := a.GetChannelByAddr(c.Peer)
+	channelByID := a.GetChannelByID(c.Number)
+	channelByPeer := a.GetChannelByAddr(c.Peer.String())
 	if channelByID != channelByPeer {
 		return errors.Errorf("You cannot use the same channel number with different peer")
 	}
 
 	// Add or refresh this channel.
 	if channelByID == nil {
+		c.allocation = a
+
 		a.channelBindingsLock.Lock()
-		defer a.channelBindingsLock.Unlock()
+		a.channelNumberBindings[c.Number] = c
+		a.channelPeerBindings[c.Peer.String()] = c
+		a.channelBindingsLock.Unlock()
 
-		c.allocation = a
-		a.channelBindings = append(a.channelBindings, c)
 		c.start(lifetime)
-
-		// Channel binds also refresh permissions.
-		a.AddPermission(NewPermission(c.Peer, a.log))
```

Comment:
Why did we remove `a.AddPermission(NewPermission(c.Peer, a.log))`, I don't remember this code at all anymore :/

Reply:
I think  `Permission` and `ChannelBind` should create with different API rather than auto create Permission by ChannelBind. I can't find this logic in RFC 5766, Maybe there is something I missed.